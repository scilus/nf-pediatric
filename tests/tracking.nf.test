nextflow_pipeline {

    options "-stub-run"

    name "Test nf-pediatric profile tracking"
    script "../main.nf"

    test("Tracking profile - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/bids/"
                params.outdir = "$outputDir"

                params.tracking = true

            }
        }

        then {
            // stable name: All files + folders in ${params.outdir}/ with a stable name.
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore:  ['pipeline_info/*.{html,json,txt}'])
            assertAll {
                assert workflow.success
                assert snapshot(
                    // Number of successfully completed tasks
                    workflow.trace.succeeded().size(),
                    // Remove the nextflow version from the versions.yml because we test it using different nextflow versions.
                    removeNextflowVersion("$outputDir/pipeline_info/nf-pediatric_software_mqc_versions.yml"),
                    // All stable name.
                    stable_name
                ).match()
            }
        }
    }

    test("Tracking profile no dwi preprocessing - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/bids/"
                params.outdir = "$outputDir"

                params.tracking = true
                params.skip_dwi_preprocessing = true

            }
        }

        then {
            // stable name: All files + folders in ${params.outdir}/ with a stable name.
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore:  ['pipeline_info/*.{html,json,txt}'])
            assertAll {
                assert workflow.success
                assert snapshot(
                    // Number of successfully completed tasks
                    workflow.trace.succeeded().size(),
                    // Remove the nextflow version from the versions.yml because we test it using different nextflow versions.
                    removeNextflowVersion("$outputDir/pipeline_info/nf-pediatric_software_mqc_versions.yml"),
                    // All stable name.
                    stable_name
                ).match()
            }
        }
    }
}
