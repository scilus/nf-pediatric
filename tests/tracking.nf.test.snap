{
    "Tracking profile - should run successfully": {
        "content": [
            39,
            {
                "ANATTODWI": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": "6.9.11"
                },
                "BETCROP_FSLBETCROP": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4",
                    "fsl": 6.0
                },
                "COREG": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": null
                },
                "CROPB0": {
                    "scilpy": "2.0.2"
                },
                "DENOISE_DWI": {
                    "mrtrix": "3.0.4"
                },
                "DENOISING_NLMEANS": {
                    "scilpy": "2.0.2"
                },
                "EXTRACTB0_RESAMPLE": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "FASTSEG": {
                    "scilpy": "2.0.0",
                    "fsl": 6.0
                },
                "N4_DWI": {
                    "scilpy": "2.0.2",
                    "N4BiasFieldCorrection": "2.4.3-gcaa60eb"
                },
                "NORMALIZE_DWI": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "PREPROC_TOPUP": {
                    "scilpy": "2.0.2",
                    "antsRegistration": null,
                    "fsl": 6.0,
                    "mrtrix": "3.0.4",
                    "imagemagick": "6.9.11"
                },
                "RECONST_DTIMETRICS": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "RECONST_FODF": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FRF": {
                    "scilpy": "2.0.2"
                },
                "RESAMPLE_DWI": {
                    "scilpy": "2.0.2"
                },
                "RESAMPLE_MASK": {
                    "scilpy": "2.0.2"
                },
                "TRACKING_LOCALTRACKING": {
                    "scilpy": "2.0.2"
                },
                "TRACKING_PFTTRACKING": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "TRACTOGRAM_MATH": {
                    "scilpy": "2.0.2"
                },
                "Workflow": {
                    "scilus/nf-pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf-pediatric_software_mqc_versions.yml",
                "sub-test",
                "sub-test/anat",
                "sub-test/anat/sub-test_desc-boundingbox.pkl",
                "sub-test/anat/sub-test_desc-brain_mask.nii.gz",
                "sub-test/anat/sub-test_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_desc-preproc_T2w.nii.gz",
                "sub-test/anat/sub-test_from-T1w_to-T2w_affine.mat",
                "sub-test/anat/sub-test_from-T1w_to-dwi_affine.mat",
                "sub-test/anat/sub-test_from-T1w_to-dwi_warp.nii.gz",
                "sub-test/anat/sub-test_from-T2w_to-T1w_affine.mat",
                "sub-test/anat/sub-test_from-dwi_to-T1w_warp.nii.gz",
                "sub-test/anat/sub-test_space-DWI_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-CSF_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-CSF_probseg.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-GM_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-GM_probseg.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-WM_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-WM_probseg.nii.gz",
                "sub-test/anat/sub-test_space-T1w_desc-preproc_T2w.nii.gz",
                "sub-test/dwi",
                "sub-test/dwi/sub-test_desc-ad.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_max.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_sum.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_total.nii.gz",
                "sub-test/dwi/sub-test_desc-brain_mask.nii.gz",
                "sub-test/dwi/sub-test_desc-concatenated_tractogram.trk",
                "sub-test/dwi/sub-test_desc-csf_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-fa.nii.gz",
                "sub-test/dwi/sub-test_desc-fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-ga.nii.gz",
                "sub-test/dwi/sub-test_desc-gm_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-local_tracking.trk",
                "sub-test/dwi/sub-test_desc-md.nii.gz",
                "sub-test/dwi/sub-test_desc-mode.nii.gz",
                "sub-test/dwi/sub-test_desc-nufo.nii.gz",
                "sub-test/dwi/sub-test_desc-peaks.nii.gz",
                "sub-test/dwi/sub-test_desc-pft_tracking.trk",
                "sub-test/dwi/sub-test_desc-preproc_b0.nii.gz",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bval",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bvec",
                "sub-test/dwi/sub-test_desc-preproc_dwi.nii.gz",
                "sub-test/dwi/sub-test_desc-rd.nii.gz",
                "sub-test/dwi/sub-test_desc-rgb.nii.gz",
                "sub-test/dwi/sub-test_desc-tensor.nii.gz",
                "sub-test/dwi/sub-test_desc-wm_fodf.nii.gz",
                "sub-test/dwi/sub-test_frf.txt",
                "sub-test/dwi/sub-test_space-DWI_label-exclude_desc-pft_probseg.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-include_desc-pft_probseg.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-seeding_desc-local_mask.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-seeding_desc-pft_mask.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-tracking_desc-local_mask.nii.gz",
                "sub-test/figures",
                "sub-test/figures/sub-test__coverage_overlay_mqc.png",
                "sub-test/figures/sub-test_gradients_mqc.png",
                "sub-test/figures/sub-test_metrics_mqc.png",
                "sub-test/figures/sub-test_tissue_segmentation_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-06T15:59:07.872668"
    },
    "Tracking profile with infant data T2 and T1 - should run successfully": {
        "content": [
            45,
            {
                "ANATTODWI": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": "6.9.11"
                },
                "APPLYTRANSFORMS": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": null
                },
                "BETCROP_SYNTHBET": {
                    "synthstrip": 1.5
                },
                "BET_DWI": {
                    "mrtrix": "3.0.4"
                },
                "CONVERT": {
                    "mrconvert": "3.0.4"
                },
                "COREG": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": null
                },
                "CROPB0": {
                    "scilpy": "2.0.2"
                },
                "CROPDWI": {
                    "scilpy": "2.0.2"
                },
                "CROPMASK": {
                    "scilpy": "2.0.2"
                },
                "DENOISE_DWI": {
                    "mrtrix": "3.0.4"
                },
                "DENOISE_REVDWI": {
                    "mrtrix": "3.0.4"
                },
                "DENOISING_NLMEANS": {
                    "scilpy": "2.0.2"
                },
                "EXTRACTB0_RESAMPLE": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "IMAGE_POWDERAVERAGE": {
                    "scilpy": "2.0.2"
                },
                "MASKS": {
                    "scilpy": "2.0.2"
                },
                "N4_DWI": {
                    "scilpy": "2.0.2",
                    "N4BiasFieldCorrection": "2.4.3-gcaa60eb"
                },
                "NORMALIZE_DWI": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "PREPROC_TOPUP": {
                    "scilpy": "2.0.2",
                    "antsRegistration": null,
                    "fsl": 6.0,
                    "mrtrix": "3.0.4",
                    "imagemagick": "6.9.11"
                },
                "RECONST_DTIMETRICS": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "RECONST_FODF": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FRF": {
                    "scilpy": "2.0.2"
                },
                "RESAMPLE_DWI": {
                    "scilpy": "2.0.2"
                },
                "RESAMPLE_MASK": {
                    "scilpy": "2.0.2"
                },
                "TISSUESEG": {
                    "mcribs": "2.1.0",
                    "ants": "2.5.3-g98bf76d",
                    "fsl": 6.0,
                    "freesurfer": "7.1.1"
                },
                "TRACKING_LOCALTRACKING": {
                    "scilpy": "2.0.2"
                },
                "Workflow": {
                    "scilus/nf-pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf-pediatric_software_mqc_versions.yml",
                "sub-test",
                "sub-test/ses-baseline",
                "sub-test/ses-baseline/anat",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-boundingbox.pkl",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-brain_mask.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-preproc_T1w.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-preproc_T2w.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-T1w_to-T2w_affine.mat",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-T2w_to-T1w_affine.mat",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-T2w_to-dwi_affine.mat",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-T2w_to-dwi_warp.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-dwi_to-T2w_warp.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_label-CSF_mask.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_label-GM_mask.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_label-WM_mask.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_space-DWI_desc-preproc_T1w.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_space-DWI_desc-preproc_T2w.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_space-T2w_desc-preproc_T1w.nii.gz",
                "sub-test/ses-baseline/dwi",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-ad.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-afd_max.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-afd_sum.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-afd_total.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-brain_mask.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-csf_fodf.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-fa.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-fodf.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-ga.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-gm_fodf.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-local_tracking.trk",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-md.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-mode.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-nufo.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-peaks.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-preproc_b0.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-preproc_dwi.bval",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-preproc_dwi.bvec",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-preproc_dwi.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-pwdavg_dwi.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-rd.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-rgb.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-tensor.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-wm_fodf.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_frf.txt",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_space-DWI_label-seeding_desc-local_mask.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_space-DWI_label-tracking_desc-local_mask.nii.gz",
                "sub-test/ses-baseline/figures",
                "sub-test/ses-baseline/figures/sub-test__coverage_overlay_mqc.png",
                "sub-test/ses-baseline/figures/sub-test_gradients_mqc.png",
                "sub-test/ses-baseline/figures/sub-test_metrics_mqc.png",
                "sub-test/ses-baseline/figures/sub-test_tissue_segmentation_mqc.png",
                "sub-test/ses-baseline/multiqc",
                "sub-test/ses-baseline/multiqc/sub-test_multiqc_data",
                "sub-test/ses-baseline/multiqc/sub-test_multiqc_plots",
                "sub-test/ses-baseline/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-06T16:06:29.569184"
    },
    "Tracking profile with infant data only T2 - should run successfully": {
        "content": [
            37,
            {
                "ANATTODWI": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": "6.9.11"
                },
                "BETCROP_SYNTHBET": {
                    "synthstrip": 1.5
                },
                "BET_DWI": {
                    "mrtrix": "3.0.4"
                },
                "CONVERT": {
                    "mrconvert": "3.0.4"
                },
                "CROPB0": {
                    "scilpy": "2.0.2"
                },
                "CROPDWI": {
                    "scilpy": "2.0.2"
                },
                "CROPMASK": {
                    "scilpy": "2.0.2"
                },
                "DENOISE_DWI": {
                    "mrtrix": "3.0.4"
                },
                "DENOISE_REVDWI": {
                    "mrtrix": "3.0.4"
                },
                "DENOISING_NLMEANS": {
                    "scilpy": "2.0.2"
                },
                "EXTRACTB0_RESAMPLE": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "IMAGE_POWDERAVERAGE": {
                    "scilpy": "2.0.2"
                },
                "MASKS": {
                    "scilpy": "2.0.2"
                },
                "N4_DWI": {
                    "scilpy": "2.0.2",
                    "N4BiasFieldCorrection": "2.4.3-gcaa60eb"
                },
                "NORMALIZE_DWI": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "PREPROC_TOPUP": {
                    "scilpy": "2.0.2",
                    "antsRegistration": null,
                    "fsl": 6.0,
                    "mrtrix": "3.0.4",
                    "imagemagick": "6.9.11"
                },
                "RECONST_DTIMETRICS": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "RECONST_FODF": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FRF": {
                    "scilpy": "2.0.2"
                },
                "RESAMPLE_DWI": {
                    "scilpy": "2.0.2"
                },
                "RESAMPLE_MASK": {
                    "scilpy": "2.0.2"
                },
                "TISSUESEG": {
                    "mcribs": "2.1.0",
                    "ants": "2.5.3-g98bf76d",
                    "fsl": 6.0,
                    "freesurfer": "7.1.1"
                },
                "TRACKING_LOCALTRACKING": {
                    "scilpy": "2.0.2"
                },
                "Workflow": {
                    "scilus/nf-pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf-pediatric_software_mqc_versions.yml",
                "sub-test",
                "sub-test/ses-baseline",
                "sub-test/ses-baseline/anat",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-boundingbox.pkl",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-brain_mask.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_desc-preproc_T2w.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-T2w_to-dwi_affine.mat",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-T2w_to-dwi_warp.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_from-dwi_to-T2w_warp.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_label-CSF_mask.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_label-GM_mask.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_label-WM_mask.nii.gz",
                "sub-test/ses-baseline/anat/sub-test_ses-baseline_space-DWI_desc-preproc_T2w.nii.gz",
                "sub-test/ses-baseline/dwi",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-ad.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-afd_max.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-afd_sum.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-afd_total.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-brain_mask.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-csf_fodf.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-fa.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-fodf.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-ga.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-gm_fodf.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-local_tracking.trk",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-md.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-mode.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-nufo.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-peaks.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-preproc_b0.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-preproc_dwi.bval",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-preproc_dwi.bvec",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-preproc_dwi.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-pwdavg_dwi.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-rd.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-rgb.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-tensor.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_desc-wm_fodf.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_frf.txt",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_space-DWI_label-seeding_desc-local_mask.nii.gz",
                "sub-test/ses-baseline/dwi/sub-test_ses-baseline_space-DWI_label-tracking_desc-local_mask.nii.gz",
                "sub-test/ses-baseline/figures",
                "sub-test/ses-baseline/figures/sub-test__coverage_overlay_mqc.png",
                "sub-test/ses-baseline/figures/sub-test_gradients_mqc.png",
                "sub-test/ses-baseline/figures/sub-test_metrics_mqc.png",
                "sub-test/ses-baseline/figures/sub-test_tissue_segmentation_mqc.png",
                "sub-test/ses-baseline/multiqc",
                "sub-test/ses-baseline/multiqc/sub-test_multiqc_data",
                "sub-test/ses-baseline/multiqc/sub-test_multiqc_plots",
                "sub-test/ses-baseline/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-06T16:03:45.359014"
    },
    "Tracking profile no dwi preprocessing - should run successfully": {
        "content": [
            33,
            {
                "ANATTODWI": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": "6.9.11"
                },
                "BETCROP_FSLBETCROP": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4",
                    "fsl": 6.0
                },
                "COREG": {
                    "ants": "2.4.3",
                    "mrtrix": "3.0.4",
                    "imagemagick": null
                },
                "DENOISING_NLMEANS": {
                    "scilpy": "2.0.2"
                },
                "EXTRACTB0_RESAMPLE": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "FASTSEG": {
                    "scilpy": "2.0.0",
                    "fsl": 6.0
                },
                "NORMALIZE_DWI": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "RECONST_DTIMETRICS": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "RECONST_FODF": {
                    "scilpy": "2.0.2"
                },
                "RECONST_FRF": {
                    "scilpy": "2.0.2"
                },
                "RESAMPLE_DWI": {
                    "scilpy": "2.0.2"
                },
                "RESAMPLE_MASK": {
                    "scilpy": "2.0.2"
                },
                "TRACKING_LOCALTRACKING": {
                    "scilpy": "2.0.2"
                },
                "TRACKING_PFTTRACKING": {
                    "scilpy": "2.0.2",
                    "mrtrix": "3.0.4"
                },
                "TRACTOGRAM_MATH": {
                    "scilpy": "2.0.2"
                },
                "Workflow": {
                    "scilus/nf-pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf-pediatric_software_mqc_versions.yml",
                "sub-test",
                "sub-test/anat",
                "sub-test/anat/sub-test_desc-boundingbox.pkl",
                "sub-test/anat/sub-test_desc-brain_mask.nii.gz",
                "sub-test/anat/sub-test_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_desc-preproc_T2w.nii.gz",
                "sub-test/anat/sub-test_from-T1w_to-T2w_affine.mat",
                "sub-test/anat/sub-test_from-T1w_to-dwi_affine.mat",
                "sub-test/anat/sub-test_from-T1w_to-dwi_warp.nii.gz",
                "sub-test/anat/sub-test_from-T2w_to-T1w_affine.mat",
                "sub-test/anat/sub-test_from-dwi_to-T1w_warp.nii.gz",
                "sub-test/anat/sub-test_space-DWI_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-CSF_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-CSF_probseg.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-GM_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-GM_probseg.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-WM_mask.nii.gz",
                "sub-test/anat/sub-test_space-DWI_label-WM_probseg.nii.gz",
                "sub-test/anat/sub-test_space-T1w_desc-preproc_T2w.nii.gz",
                "sub-test/dwi",
                "sub-test/dwi/sub-test_desc-ad.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_max.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_sum.nii.gz",
                "sub-test/dwi/sub-test_desc-afd_total.nii.gz",
                "sub-test/dwi/sub-test_desc-brain_mask.nii.gz",
                "sub-test/dwi/sub-test_desc-concatenated_tractogram.trk",
                "sub-test/dwi/sub-test_desc-csf_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-fa.nii.gz",
                "sub-test/dwi/sub-test_desc-fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-ga.nii.gz",
                "sub-test/dwi/sub-test_desc-gm_fodf.nii.gz",
                "sub-test/dwi/sub-test_desc-local_tracking.trk",
                "sub-test/dwi/sub-test_desc-md.nii.gz",
                "sub-test/dwi/sub-test_desc-mode.nii.gz",
                "sub-test/dwi/sub-test_desc-nufo.nii.gz",
                "sub-test/dwi/sub-test_desc-peaks.nii.gz",
                "sub-test/dwi/sub-test_desc-pft_tracking.trk",
                "sub-test/dwi/sub-test_desc-preproc_b0.nii.gz",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bval",
                "sub-test/dwi/sub-test_desc-preproc_dwi.bvec",
                "sub-test/dwi/sub-test_desc-preproc_dwi.nii.gz",
                "sub-test/dwi/sub-test_desc-rd.nii.gz",
                "sub-test/dwi/sub-test_desc-rgb.nii.gz",
                "sub-test/dwi/sub-test_desc-tensor.nii.gz",
                "sub-test/dwi/sub-test_desc-wm_fodf.nii.gz",
                "sub-test/dwi/sub-test_frf.txt",
                "sub-test/dwi/sub-test_space-DWI_label-exclude_desc-pft_probseg.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-include_desc-pft_probseg.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-seeding_desc-local_mask.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-seeding_desc-pft_mask.nii.gz",
                "sub-test/dwi/sub-test_space-DWI_label-tracking_desc-local_mask.nii.gz",
                "sub-test/figures",
                "sub-test/figures/sub-test__coverage_overlay_mqc.png",
                "sub-test/figures/sub-test_gradients_mqc.png",
                "sub-test/figures/sub-test_metrics_mqc.png",
                "sub-test/figures/sub-test_tissue_segmentation_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-06T16:01:09.833035"
    }
}